tests/test_fetcher_ng.py

```diff
@@ -94,6 +97,40 @@ def test_fetch_in_chunks(self) -> None:
             # Check that we calculate chunks as expected
             self.assertEqual(chunks_count, expected_chunks_count)
 
+    # Fetch data with Content-Encoding gzip (or deflate)
+    def test_fetch_content_encoding(self):
+        """
+        Regression test for issue #2047
+
+        By default, requests/urllib3 will automatically try to decode data
+        that is served with a "Content-Encoding: gzip" header (or "deflate").
+        As the length of the decoded data typically would be different from
+        the expected number of bytes specified in the TUF targets metadata,
+        this would give rise to a DownloadLengthMismatchError later on. Thus,
+        we need to ensure that a file served with the "Content-Encoding"
+        header is *not* decoded by the RequestsFetcher.
+        """
+        # Serve dummy file with "Content-Encoding: gzip" header
+        content_encoding_header = json.dumps({"Content-Encoding": "gzip"})
+        headers = {utils.REQUEST_RESPONSE_HEADERS: content_encoding_header}
+        get_with_headers = partialmethod(requests.Session.get, headers=headers)
```

Comment:
As I am new to `functools` and `partialmethod` what is the bonus of creating a `partialmethod` object instead of just calling `response.Session.get` directly in line 123 with the headers you want?